assembly/runtime/promise.ts

```diff
@@ -0,0 +1,145 @@
+import { util } from "./util";
+import { env } from "./env";
+import { u128 } from ".";
+
+function encodeArgs<T>(t: T) : Uint8Array {
+  if (t instanceof Uint8Array) {
+    return t;
+  }
+  return encode<T>(t);
+}
+
+// TODO: not sure if these types make any sense at all
+export type Balance = u128;
+export type Gas = u64;
+export type PublicKey = Uint8Array;
+export type AccountId = Uint8Array;
+
+export type CreateAccount = "CreateAccount"; // TODO: this seems wierd
+export type DeployContract = { code: Uint8Array };
+export type FunctionCall = { methodName: Uint8Array, args: Uint8Array, amount: Balance, gas: Gas };
+export type Transfer = { amount: Balance };
+export type Stake = { amount: Balance, publicKey: PublicKey };
+export type AddFullAccessKey = { publicKey: PublicKey };
+export type AddAccessKey = { publicKey: PublicKey, allowance: Balance, receiverId: AccountId, methodNames: Uint8Array[] };
+export type DeleteKey = { publicKey: PublicKey };
+export type DeleteAccount = { beneficiaryId: AccountId };
+
+export type Action =
```

Comment:
AssemblyScript doesn't support Union types yet.  Only `T | null` for nullable types.

Reply:
removed